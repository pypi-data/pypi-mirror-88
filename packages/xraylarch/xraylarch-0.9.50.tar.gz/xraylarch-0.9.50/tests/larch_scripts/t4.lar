#print '---- TDL test 1'
x = sqrt(11)
y = cos(x)

print 'x = sqrt(11) = ', x
print 'y = cos(x)   = ',y

# n = 1
# while n<8:
#     print ' Loop', n
#     n = n + 1
# #endwhile

x = 3
_main.x = 4.0
print  ' x = ', x, _main.x
def f1(a,b,c=1):
    print  ' this is f1 ', a, b,c 
    if c < 0:  c = -c
    x = a*b*c
    # print show_group('_main')
 
    print ' in f1 : x = %f, _main.x = %f' % (x, _main.x)
    print " error at function f1:: q= ",  a/(_main.x + 4)
    return x
enddef

c0xx  = 3.4
# print 'c0xx = ' , c0xx

a = 2
b = 3

y = f1(a,sqrt(b/a),c=2.01)

print "after calling f1, y = %f, x=%f" % (y,x)
# 
print ' and now a runtime error:'
print  3 + f1(a,2,c=10)
print  3 + f1(a,2,d=10)

print 'X"'
