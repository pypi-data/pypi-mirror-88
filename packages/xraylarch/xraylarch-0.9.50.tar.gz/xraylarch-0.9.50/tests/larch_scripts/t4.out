x = sqrt(11) =  3.31662479036
y = cos(x)   =  -0.984720943124
 x =  4.0 4.0
 this is f1  2 1.22474487139 2.01
 in f1 : x = 4.923474, _main.x = 4.000000
 error at function f1:: q=  0.25
after calling f1, y = 4.923474, x=4.000000
 and now a runtime error:
 this is f1  2 2 10
 in f1 : x = 40.000000, _main.x = 4.000000
 error at function f1:: q=  0.25
43
== Error ==
extra keyword arguments for Procedure f1 (d):  t4.lar, line number 17
     def f1(a,b,c=1):
 t4.lar, line number 39
     print  3 + f1(a,2,d=10)
===========


